File: app/src/main/java/com/hitesh/todopod/ItemsViewModel.java
package com.hitesh.todopod;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import java.util.List;

public class ItemsViewModel extends AndroidViewModel {
    private itemsRepository repository;
    private LiveData<List<items>> allitemss;
    public ItemsViewModel(@NonNull Application application) {
        super(application);
        repository = new itemsRepository(application);
        allitemss = repository.getAllitemss();
    }
    public void insert(items items) {
        repository.insert(items);
    }
    public void update(items items) {
        repository.update(items);
    }
    public void delete(items items) {
        repository.delete(items);
    }
    public void deleteAllitemss() {
        repository.deleteAllitemss();
    }
    public LiveData<List<items>> getAllitemss() {
        return allitemss;
    }
}

File: app/src/main/java/com/hitesh/todopod/model/ItemsViewModel.kt
package com.hitesh.todopod.model

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import com.hitesh.todopod.helper.ItemsRepository
import com.hitesh.todopod.items

class ItemsViewModel(application: Application) : AndroidViewModel(application) {
    private val repository: ItemsRepository = ItemsRepository(application)
    val allitems: LiveData<List<items>>

    fun insert(items: items?) {
        repository.insert(items)
    }

    fun update(items: items?) {
        repository.update(items)
    }

    fun delete(items: items?) {
        repository.delete(items)
    }

    fun deleteAllitems() {
        repository.deleteAllitems()
    }

    init {
        allitems = repository.getAllitems()
    }
}